pandas/tests/indexes/interval/test_interval_range.py

```diff
@@ -220,6 +220,27 @@ def test_float_subtype(self, start, end, freq):
         expected = "int64" if is_integer(start + end) else "float64"
         assert result == expected
 
+    @pytest.mark.parametrize(
+        "iteration, start, end",
+        [
+            (0, np.int8(1), np.int8(10)),
+            (1, np.int8(1), np.float16(10)),
+            (2, np.float32(1), np.float32(10)),
+            (3, 1, 10),
+            (4, 1, 10.0),
+        ],
+    )
+    def test_interval_dtype(self, iteration, start, end):
+        result = interval_range(start=start, end=end).dtype.subtype
+        expected = [
```

Comment:
Can you put this in the `parametrize`?

Reply:
You mean put expected in the paramaterize?

Reply:
Please check the new changes